### DISCUSSION TITLE
GIS support?

### DISCUSSION THREAD

**Discussion by nrbnlulu** [slint-ui/slint#2631] [Q&A]
Is there any GIS support something like [QML Map](https://doc.qt.io/qt-6/qml-qtlocation-map.html#details)? what would be the alternative?

**Comment by tronical** [comment]
What programming language are you targeting?

**Reply by nrbnlulu** [reply]
rust

**Reply by tronical** [reply]
Hmm. I don't think that there's an out-of-the-box solution as-is. I see that the maplibre project at https://github.com/maplibre/maplibre-rs is actively working on good Rust bindings. It appears that their API is designed to take over a winit window and render the map into it with wgpu. We don't support rendering on top of wgpu (yet), but with the access that maplibre provides, you could probably instantiate Slint's software renderer with a MinimalSoftwareWindow, render the UI into a buffer, and then upload that into a texture, and add it somehow to the maplibre scene.
